src/main/java/net/spy/memcached/MemcachedClient.java

```diff
@@ -255,12 +255,18 @@ private MemcachedClient(ConnectionFactory cf, List<InetSocketAddress> addrs, boo
         String hostName = addrs.get(0).getHostName();
         //All config endpoints has ".cfg." subdomain in the DNS name.
         if(hostName != null && hostName.contains(".cfg.")){
-          cf = new DefaultConnectionFactory(ClientMode.Dynamic);
+          if (cf == null) {
+            cf = new DefaultConnectionFactory(ClientMode.Dynamic);
+          } else {
+            cf.setClientMode(ClientMode.Dynamic);
+          }
```

Comment:
Re-factor this method into a private method. Something like `ConnectionFactory updateClientMode(connectionFactory f, ClientMode mode)` maybe? That way you can re-use this function in the fallback logic below.